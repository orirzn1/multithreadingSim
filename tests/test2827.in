L16
S6
O6
N5

T0
I@0x00000000
LOAD $1, $0, $7
ADDI $7, $2, 0x4
ADD $5, $7, $7
SUBI $3, $1, 0x1
SUBI $3, $2, 0x4
HALT $0

T1
I@0x00000000
ADD $1, $7, $4
LOAD $6, $7, $5
ADD $1, $6, $7
ADD $3, $5, $7
SUBI $4, $0, 0x3
SUB $4, $1, $3
HALT $0

T2
I@0x00000000
ADDI $5, $6, 0x2
STORE $0, $0, 0x2
ADD $2, $3, $4
ADDI $7, $7, 0x3
LOAD $6, $0, 0x4
LOAD $4, $2, 0x2
SUB $4, $7, $4
ADDI $0, $5, 0x2
ADD $3, $0, $7
SUB $1, $3, $3
HALT $0

T3
I@0x00000000
SUB $0, $0, $6
ADDI $6, $0, 0x3
SUBI $3, $0, 0x1
STORE $3, $2, 0x3
LOAD $0, $2, $7
SUB $0, $6, $0
SUBI $4, $2, 0x2
HALT $0

T4
I@0x00000000
ADD $7, $1, $3
SUB $7, $7, $1
SUB $1, $7, $0
SUBI $5, $3, 0x2
SUBI $5, $2, 0x1
LOAD $0, $5, $7
STORE $5, $7, $5
SUBI $0, $1, 0x2
STORE $2, $6, 0x4
HALT $0

D@0x0
0x9
0x8
0x9
0x3
0x6
0x2
0x8
0x7
