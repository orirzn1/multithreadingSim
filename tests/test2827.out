
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x9	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x8	R6 = 0x0	R7 = 0x4
Register file thread id 1:
	R0 = 0x0	R1 = 0x9	R2 = 0x0	R3 = 0x0	R4 = 0x9	R5 = 0x0	R6 = 0x9	R7 = 0x0
Register file thread id 2:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x7	R4 = 0x3	R5 = 0x2	R6 = 0x8	R7 = 0x3
Register file thread id 3:
	R0 = 0x3	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFE	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 4:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 3.047619

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x8	R6 = 0x0	R7 = 0x4
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x7	R4 = 0x3	R5 = 0x2	R6 = 0x0	R7 = 0x3
Register file thread id 3:
	R0 = 0x3	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFE	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 4:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.357143

